includes/class-llms-order-generator.php

```diff
@@ -0,0 +1,733 @@
+		$data['user_id'] = $this->student ? $this->student->get( 'id' ) : '';
+
+		return $data;
+
+	}
+
+	/**
+	 * Performs all required data validations necessary to create the order.
+	 *
+	 * @since [version]
+	 *
+	 * @param boolean $validate_order Whether or not order data should be validated. This is `true` when running `confirm()` and `false` otherwise.
+	 * @return boolean|WP_Errors Returns `true` if all validations pass or an error object.
+	 */
+	protected function validate( $validate_order = false ) {
+
+		/**
+		 * Allows 3rd party validation prior to generation of an order.
+		 *
+		 * This validation hooks runs prior to all default validation.
+		 *
+		 * @since [version]
+		 *
+		 * @param null|WP_Error $validation_errors Halts checkout and returns the supplied error.
+		 */
+		$before_validation = apply_filters( 'llms_before_generate_order_validation', null );
+		if ( is_wp_error( $before_validation ) ) {
+			return $before_validation;
+		}
+
+		$validations = array(
+			'validate_plan',
+			'validate_coupon',
+			'validate_gateway',
+			'validate_terms',
+			'validate_user',
+		);
+
+		if ( $validate_order ) {
+			array_unshift( $validations, 'validate_order' );
+		}
+
+		foreach ( $validations as $func ) {
+			$res = $this->{$func}();
+			if ( is_wp_error( $res ) ) {
+				return $res;
+			}
+		}
+
+		/**
+		 * Allows 3rd party prior to generation of an order.
+		 *
+		 * This validation hooks runs after to all default validation passes.
+		 *
+		 * @since [version]
+		 *
+		 * @param boolean|WP_Error $validation_errors Halts checkout and returns the supplied error.
+		 */
+		return apply_filters( 'llms_after_generate_order_validation', true );
+
+	}
+
+	/**
+	 * Validates the coupon.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean|WP_Error Returns `true` on success or an error object.
+	 */
+	protected function validate_coupon() {
+
+		// If a coupon is being used, validate it.
+		if ( ! empty( $this->data['llms_coupon_code'] ) ) {
+
+			$code = sanitize_text_field( $this->data['llms_coupon_code'] );
+
+			// Locate the coupon post ID.
+			$coupon_id = llms_find_coupon( $code );
+			if ( ! $coupon_id ) {
+				return $this->error(
+					self::E_COUPON_NOT_FOUND,
+					sprintf(
+						// Translators: %s = The user-submitted coupon code.
+						__( 'Coupon code "%s" not found.', 'lifterlms' ),
+						$code
+					)
+				);
+			}
+
+			// Validate the coupon for the current plan.
+			$coupon = llms_get_post( $coupon_id );
+			$valid  = $coupon->is_valid( $this->plan->get( 'id' ) );
+			if ( is_wp_error( $valid ) ) {
+				return $this->error( self::E_COUPON_INVALID, $valid->get_error_message() );
+			}
+
+			$this->coupon = $coupon;
+
+		}
+
+		return true;
+	}
+
+	/**
+	 * Validates the payment gateway.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean|WP_Error Returns `true` on success or an error object.
+	 */
+	protected function validate_gateway() {
+
+		$coupon_id = $this->coupon ? $this->coupon->get( 'id' ) : null;
+
+		/**
```

Comment:
```suggestion
		/*
```
It's a [multi-line comment](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments), not a DocBlock.